src/core/public/chrome/ui/header/header_search_bar.tsx

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButtonIcon,
+  EuiFieldSearch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiPanel,
+  EuiPopover,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+import React, { ReactNode, useRef, useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { GlobalSearchCommand, SearchObjectTypes } from '../../global_search';
+
+interface Props {
+  globalSearchCommands: GlobalSearchCommand[];
+  panel?: boolean;
+  onSearchResultClick?: () => void;
+}
+
+/**
+ * search input match with `@` will handled by saved objects search command
+ * search input match with `>` will handled by plugin customized commands
+ */
+export const SAVED_OBJECTS_SYMBOL = '@';
+export const COMMANDS_SYMBOL = '>';
```

Comment:
It looks a little bit weird to me. Why don't we just aggregate all the results from `globalSearchCommands`?

Reply:
That's to filter what kind of searchCommand we want to run. If user input is start with `@` like  `@flights` it will only search for saved objects title has flights.

Reply:
Oh, that's a big vision in the future but seems weird that we implement in this PR. I think these should be done in an enhancement PR if we really have the use case.

Reply:
Agree these should come with future enhancement PR, these are built in symbols for saved objects and commands, so i put it here for in advance.

regarding to prefix registration, that's might be long term plan if we plan to support more command types.